pipeline/workflow-builder-dev.Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM quay.io/kiegroup/kogito-swf-builder-nightly:main-2024-04-08 AS builder
+FROM quay.io/kiegroup/kogito-swf-builder-nightly:latest AS builder
```

Comment:
latest isn't recommended by konflux, let's pin to a specific version

Reply:
yes but this is for `dev` and IMO, for dev purposes it's OK to use the latest to test changes, no?

Reply:
Somehow I remembered that konflux has no-latest-tag tolerance. if that isn't the case, that's fine.
but giving it another thought - we won't be able to know against which builder image the workflow was created.
that should provide us some information when debugging issues or when trying to reproduce one.

Reply:
> that should provide us some information when debugging issues or when trying to reproduce one.

the dev image should only be used when developing something on the workflow, so that does no apply here

> we won't be able to know against which builder image the workflow was created.

When someone is developing (s)he knows what's going on

Reply:
I don't mind chaning that dockedrfile, it is not in use by konflux or the the github action. Just looking more closely I see that there is no meaningful difference to the workflow-builder.Dockerfile other than the builder image.  In that case why not remove this file altogether and use a build-arg if you want some latest? or even create a make rule for building for devlopers?

Reply:
There used to be differences between the two. If this is no longer the case (when we had the extensions included in the build image of the d/s builder image).
Wouldn't the developer will need to have a dockerfile, even with argfile? where do you recommend to have that file for him? (talking about upstream contributor/developer)